daemon/api_users_test.go

```diff
@@ -223,13 +228,48 @@ func (s *userSuite) TestPostUserBadAction(c *check.C) {
 	c.Check(rsp, check.DeepEquals, BadRequest(`unsupported user action "patatas"`))
 }
 
-func (s *userSuite) TestPostUserActionRemoveNotImplemented(c *check.C) {
+func (s *userSuite) TestPostUserActionRemoveNoUsername(c *check.C) {
 	buf := bytes.NewBufferString(`{"action":"remove"}`)
 	req, err := http.NewRequest("POST", "/v2/users", buf)
 	c.Assert(err, check.IsNil)
 
 	rsp := postUsers(usersCmd, req, nil).(*resp)
-	c.Check(rsp, check.DeepEquals, NotImplemented("not implemented"))
+	c.Check(rsp, check.DeepEquals, BadRequest("need a username to remove"))
+}
+
+func (s *userSuite) TestPostUserActionRemoveNoUserInState(c *check.C) {
+	called := false
```

Comment:
(nitpick^2) this could be a counter, this way we enure it's not called multiple times by accident.